roboteam_ai/src/coach/defence/DefencePositionCoach.cpp

```diff
@@ -309,7 +309,7 @@ world::WorldData DefencePositionCoach::getTheirAttackers(const world::WorldData
     for (const world::Robot &robot :world.them) {
         // we remove any attackers that are outside of the field or in our defence area
         if (! world::field->pointIsInDefenceArea(robot.pos, true, 0.04)
-                && world::field->pointIsInField(robot.pos, 0.1)) {
+                && world::field->pointIsInField(robot.pos, -0.1)) {
```

Comment:
If I want to handle robots that are out of the field, specifically behind our defense line, this creates a lot of edge cases in the calculations on lower levels which complicate the code by a lot. I chose not to include these because in 99% of the cases these are not the robots creating most danger regardless as they have no direct shot to goal or are very far away. Same goes for why I am checking if the point is in defence area as this is uncoverable.